### PR TITLE
neovim

### PR DESCRIPTION
**Pull request by jerowe**
Neovim is a refactor—and sometimes redactor—in the tradition of
    Vim, which itself derives from Stevie. It is not a rewrite but a continuation
    and extension of Vim. Many clones and derivatives exist; some are very clever,
    but none are Vim. Neovim strives to be a superset of Vim except for some
    intentionally-removed misfeatures. It is built for users who want the good
    parts of Vim, and more.

### CODE REVIEW

**Comment by jerowe** [comment]
I'm getting some odd errors with travis.

```
Traceback (most recent call last):

  File "bootstrap-obvious-ci-and-miniconda.py", line 103, in <module>

    install_obvci=not args.without_obvci)

  File "bootstrap-obvious-ci-and-miniconda.py", line 78, in main

    subprocess.check_call(cmd)

  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/subprocess.py", line 540, in check_call

    raise CalledProcessError(retcode, cmd)

subprocess.CalledProcessError: Command '['bash', 'Miniconda3-latest-MacOSX-x86_64.sh', '-b', '-p', '/Users/travis/miniconda']' returned non-zero exit status 2

./.CI/build_all: line 15: conda: command not found

./.CI/build_all: line 16: conda: command not found

./.CI/build_all: line 17: conda: command not found

./.CI/build_all: line 18: conda: command not found

./.CI/build_all: line 19: run_conda_forge_build_setup: No such file or directory

Finding recipes merged in master and removing them from the build.

./.CI/build_all: line 24: /usr/local/bin/git: No such file or directory

./.CI/build_all: line 28: conda-build-all: command not found
```

**Comment by jakirkham** [comment]
Trying merging with or rebasing on `master`. We have made a few important changes recently.

**Comment by jerowe** [comment]
Alright. We will let that run.

I had a case of too many projects and not enough me the last few months, but I should be able to get back to this and the sqlite with readline feedstock support. ;-)

**Comment by jerowe** [comment]
@jakirkham , any idea why the travis build is insisting on glibtoolize?

https://github.com/joyent/libuv/issues/1200

Its stated here that if I just export LIBTOOLIZE=libtoolize that it should be fine, and that looks to be the case, but I am still getting glibtoolize.

**Comment by jakirkham** [comment]
Could try adding a symlink to `glibtoolize` in `PREFIX/bin` that points to `libtoolize`? Would definitely raise an issue upstream though as ignoring the environment variable is not cool.

**Comment by jerowe** [comment]
I just don't get it.

```
Try 'glibtool --help' for more information.

glibtool:   error: unrecognised option: '--mode'
```

On my local condaforge docker image I can create a symlink from libtool to glibtool, and glibtool --mode is just fine.

**Comment by jerowe** [comment]
@jakirkham, do you have any idea what this means?

```
==== Building LuaJIT 2.0.4 ====

HOSTCC    host/minilua.o

HOSTLINK  host/minilua

DYNASM    host/buildvm_arch.h

dyld: Library not loaded: @rpath/libiconv.2.dylib

  Referenced from: /Users/travis/miniconda/conda-bld/work/neovim-0.1.5/.deps/build/src/luajit/src/host/minilua

  Reason: Incompatible library version: minilua requires version 8.0.0 or later, but libiconv.2.dylib provides version 7.0.0

make[5]: *** [host/buildvm_arch.h] Trace/BPT trap: 5

make[4]: *** [src/luajit] Error 2

make[3]: *** [build/src/luajit-stamp/luajit-install] Error 2

make[2]: *** [CMakeFiles/luajit.dir/all] Error 2

make[1]: *** [all] Error 2

make: *** [deps] Error 2
```

**Comment by frol** [comment]
@jerowe I have just accidentally bumped into this PR because I was building libuv recipe (#2031) and had the "missing glibtoolize" issue. It seems that I have succeeded with building it for Linux and Mac OS, so if it is merged, you may just set `libuv` as a build/run dependency here.

**Comment by jerowe** [comment]
@frol, great, thanks! I see that the libuv passes, so I will keep an eye on it and try again.

File: recipes/neovim/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+build:
+  number: 0
+  skip: True  #[win]
+  rpaths:
+    - lib/
+
+requirements:
+  build:
+      - gcc # [linux]
+      - llvm # [osx]
```
Comment (jakirkham) [inline recipes/neovim/meta.yaml]: Would drop these.

File: recipes/neovim/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      - autoconf
+      - libiconv
+      - readline
+      - ncurses
+      - unzip
+      - pkgconfig
+      - libuv
+      - perl
+  run:
+      - libgcc # [linux]
```
Comment (jakirkham) [inline recipes/neovim/meta.yaml]: Also this.

File: recipes/neovim/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  url: https://github.com/{{name}}/{{name}}/archive/v{{version}}.tar.gz
+  sha256: {{ sha256 }}
+  patches:
+      - Makefile.patch
+
+build:
+  number: 0
+  skip: True  #[win]
+  rpaths:
+    - lib/
```
Comment (jakirkham) [inline recipes/neovim/meta.yaml]: I'd recommend dropping this for now. Could be complicating things.

File: recipes/neovim/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,72 @@
+{% set name = "neovim" %}
+{% set version = "0.1.7" %}
+{% set sha256 = "d8f885d019b1ad608f36ae23b8f1b15b7e33585e16f7514666ab6c9809bb4b7e" %}
+
+package:
+  name: neovim
+  version: '0.1.7'
```
Comment (jakirkham) [inline recipes/neovim/meta.yaml]: Should be `{{ version }}`.

File: recipes/neovim/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+      - cmake
+      - m4
+      - libtool
+      - automake
+      - autoconf
+      - libiconv
+      - readline
+      - ncurses
+      - unzip
+      - pkgconfig
```
Comment (jakirkham) [inline recipes/neovim/meta.yaml]: Probably want `pkg-config` (note `-`).

(The other is an unfortunately named Python package. 😒)

File: recipes/neovim/meta.yaml [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+      - libuv
+      - perl
+  run:
+      - libgcc # [linux]
+      - zlib
+      - libiconv
+      - readline
+      - ncurses
+      - pkgconfig
```
Comment (jakirkham) [inline recipes/neovim/meta.yaml]: Guessing this is not needed at run time.

File: recipes/neovim/meta.yaml [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+source:
+  fn: {{ name }}-{{ version }}.tar.gz
+  url: https://github.com/{{name}}/{{name}}/archive/v{{version}}.tar.gz
+  sha256: {{ sha256 }}
+  patches:
+      - Makefile.patch
+
+build:
+  number: 0
+  skip: True  #[win]
```
Comment (jakirkham) [inline recipes/neovim/meta.yaml]: nit: Please add a space after `#` and before `[`.

**Comment by jakirkham** [comment]
Might help to start pinning down dependencies based on this [list]( https://github.com/conda-forge/conda-forge.github.io/blob/724daeb0250db4c5de739f809f7b14b7d437b029/scripts/pin_the_slow_way.py#L29-L70 ). Both to get it ready to go and get rid of any other confounding factors.

File: recipes/neovim/meta.yaml (hunk 1 above)
Comment (jakirkham) [inline recipes/neovim/meta.yaml]: Also would add `toolchain` here.

File: recipes/neovim/meta.yaml [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+      - libtool
+      - automake
+      - autoconf
+      - libiconv
+      - readline
+      - ncurses
+      - unzip
+      - pkgconfig
+      - libuv
+      - perl
```
Comment (jakirkham) [inline recipes/neovim/meta.yaml]: Also would just pin this to an exact version of `perl`. Take a look at our feedstock and just use latest if it works.

File: recipes/neovim/meta.yaml [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+  run:
+      - libgcc # [linux]
+      - zlib
+      - libiconv
+      - readline
+      - ncurses
+      - pkgconfig
+      - libuv
+      - perl
```
Comment (jakirkham) [inline recipes/neovim/meta.yaml]: Do we need this at run time or just for the build?

File: recipes/neovim/meta.yaml [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+
+test:
+  commands:
+    - nvim --version
+
+about:
+  home: 'https://github.com/neovim/neovim'
+  license: 'Apache 2.0'
+  license_file: LICENSE
```
Comment (Ciornenichii) [inline recipes/neovim/meta.yaml:58]: Ciornenichii stepan

**Pull request closed without merging**